feat: added plugin-twilio this can send message using sms or whatsapp

juanc07 opened the pull request:
<!-- Use this template by filling in information and copying and pasting relevant items out of the HTML comments. -->

# Relates to

<!-- LINK TO ISSUE OR TICKET -->

<!-- This risks section must be filled out before the final review and merge. -->

# Risks

<!--
Low, we just need to follow the format when you want to send SMS or WhatsApp message, please read the read ME about this pattern
-->

# Background

## What does this PR do?
this will add a twilio api implemenation for send SMS or WhatsApp Message

## What kind of change is this? this is a new plugin so this will be addition to current repo

<!--
Bug fixes (non-breaking change which fixes an issue)
Improvements (misc. changes to existing features)
Features (non-breaking change which adds functionality)
Updates (new versions of included code)
-->
this is none breaking, all possible errors are catch

<!-- This "Why" section is most relevant if there are no linked issues explaining why. If there is a related issue, it might make sense to skip this why section. -->
<!--
## Why are we doing this? Any context or related work?
-->
i saw Shaw post this in X with reward so i did this for that reward, i really need money :)

# Documentation changes needed?

<!--
My changes do not require a change to the project documentation.
My changes require a change to the project documentation.
If documentation change is needed: I have updated the documentation accordingly.
-->
it need some small changes you need to update Agent package.json to include plugin-twilio, then wire it in agent for plugin initialization and then pnpm install and pnpm build just to make sure

<!-- Please show how you tested the PR. This will really help if the PR needs to be retested and probably help the PR get merged quicker. -->

# Testing
for testing you just need to tell the AI Agent the exact command format, please check the Usage Sample 

# Usage Sample

The message that you want to send must be inside a Single Quote or double quotes

Example 1:
Please send sms to [phone number], and my message is '[your message here]'
Please send whats app message to [phone number], and my message is '[your message here]'

Example 2:
Please send sms to [phone number], and my message is "[your message here]"
Please send whats app message to [phone number], and my message is "[your message here]"

## Where should a reviewer start?
inside the plugin-twilio to make sure i follow the standards

## Detailed testing steps

add this to .env file and fill in the details

# Twilio Part
TWILIO_ACCOUNT_SID=
TWILIO_AUTH_TOKEN=
TWILIO_PHONE_NUMBER=
TWILIO_WHATSAPP_PHONE_NUMBER=

then follow this steps 

# How to use
1. create your .env file , if you don't have it yet and then populate it with the information above, make sure to fill in all information
2. Add this project into your eliza os project under packages
3. using terminal go inside plugin-twilio then type pnpm install twilio
4. go inside your agent folder update the package.json add this "@elizaos/plugin-twilio": "workspace:*"
5. add this inside your Agent index.ts import { twilioPlugin } from "@elizaos/plugin-twilio";
6. Add twilioPlugin in Agent Runtime still inside Agent index.ts
7. pnpm install
8. pnpm build
9. pmpn start --character="characters/nameofyouragentcharacterfile.character.json"

#Note: Make sure you have the following:
1. Verified Twilio developer account
2. Verified Twilio phone number
3. Verified Twilio Whatspapp enabled phone number
4. have enough credits but they provide free small credits when your account is new

visit twilio: https://www.twilio.com
twilio quick start guides: https://www.twilio.com/docs/messaging/quickstart
twilio documentation: https://www.twilio.com/docs/messaging
Free Trial Account: https://www.twilio.com/docs/messaging/guides/how-to-use-your-free-trial-account

# For WhatsApp guides follow the link below you need to have a separate twilio whatsapp enabled phone number 
https://www.twilio.com/docs/whatsapp
https://www.twilio.com/docs/whatsapp/quickstart/node
https://www.twilio.com/docs/whatsapp/getting-started#registering-a-whatsapp-sender
https://www.twilio.com/docs/verify/whatsapp


#Some Other Whats App Info that you might need
https://www.twilio.com/docs/whatsapp

#Twilio Phone Number guidelines
https://www.twilio.com/en-us/guidelines


# Clarification this project is intended to be use/place inside elizaos project packages, this can't work alone

# Sample implementation can be found here 
https://github.com/juanc07/AgentSoulSpark


<!--
None: Automated tests are acceptable.
-->

<!--
- As [anon/admin], go to [link]
  - [do action]
  - verify [result]
-->

<!-- If there is a UI change, please include before and after screenshots or videos. This will speed up PRs being merged. It is extra nice to annotate screenshots with arrows or boxes pointing out the differences. -->
<!--
## Screenshots
### Before
### After
-->

<!-- If there is anything about the deployment, please make a note. -->
<!--
# Deploy Notes
-->

<!--  Copy and paste command line output. -->
![logs-twilio-send-sms](https://github.com/user-attachments/assets/8959aad0-10a3-427b-bef7-0b7b51f19e28)
![Screenshot_20250103_141851_Messages](https://github.com/user-attachments/assets/97450572-1ba9-44d6-9bac-36189f822b45)

<!--
## Database changes
-->

<!--  Please specify deploy instructions if there is something more than the automated steps. -->
<!--
## Deployment instructions
-->

<!-- If you are on Discord, please join https://discord.gg/ai16z and state your Discord username here for the contributor role and join us in #development-feed -->

<!--
## Discord username
this is my discord name  warfreakzplays
-->

marcellodesales committed:
:bug: generating package-lock.json for docusaurus updates

marcellodesales committed:
:whale: Just changing Dockerfile.docs with build comments

marcellodesales committed:
:scroll: update package-lock.json with current docs package.json

marcellodesales committed:
:scroll: update pnpm-lock.yaml with docusaurus changes

odilitime commented on agent/src/index.ts:
```diff
@@ -525,6 +526,7 @@ export async function createAgent(
         // character.plugins are handled when clients are added
         plugins: [
             bootstrapPlugin,
+            twilioPlugin,
```
should this be gated by required secrets for twilio?

hellopleasures commented:
ph represent, this is nice, salamat juan for this!!!

rawgroundbeef committed:
fix(solana token provider): await cached data

mcp97 committed:
Merge pull request #1829 from bitcoinbender/patch-1

mcp97 committed:
Merge branch 'develop' into refactor/readme

affaan-m committed:
chore: update bootstrap plugin export

Relates to:
Plugin export collisions in character files

Risks
Low - This is a minor bug fix that ensures proper module exports

Background
What does this PR do?
Ensures the bootstrap plugin has a proper default export at the bottom to avoid collisions when used in the actions module in character files. This maintains consistency with other plugins in the codebase that follow the same pattern (like the EVM plugin).

What kind of change is this?
Bug fixes (non-breaking change which fixes an issue)

Documentation changes needed?
My changes do not require a change to the project documentation.

Testing
Where should a reviewer start?
Check packages/plugin-bootstrap/src/index.ts to verify the proper export structure
Verify that the plugin can be properly imported in character files without collisions
Detailed testing steps
Load a character file that uses the bootstrap plugin
Verify that no export collisions occur when the plugin is loaded
Verify that all plugin actions are properly accessible
Deploy Notes
No special deploy steps required. This is a straightforward bug fix that maintains backward compatibility.

Discord:
affaanmustafa

mcp97 committed:
Merge branch 'develop' into develop

ai16z-demirix committed:
improvement: replacing console.log with elizaLogger.

ai16z-demirix committed:
Merge branch 'develop' into improvement/eliza-logger

web3isthefuture committed:
Update README_KOR.md

Modified the previous sentence to align with the meaning in Korean.

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge pull request #1815 from gmh5225/develop

mcp97 committed:
Merge branch 'develop' into improvement/eliza-logger

mcp97 committed:
Merge pull request #1834 from ai16z-demirix/improvement/eliza-logger

mcp97 committed:
Merge branch 'develop' into main

mcp97 committed:
Merge branch 'develop' into refactor/readme

ai16z-demirix committed:
Revert "refactor: (draft) improving API error handling for coinbase integration"

This reverts commit 3a9e84e538b9253d567aa32bde3bb9d7b67ba274.

Commit:
correct faq docs link in both contribution md files

mcp97 committed:
Merge branch 'develop' into patch-4

ai16z-demirix committed:
test: fixing generation.test.ts. Adding tests for memory, goals and provider

ai16z-demirix committed:
Merge branch 'develop' into test/goals-memory-provider

MacsDickinson committed:
Merge branch 'develop' into 1838--fix-faq-link-in-contributing-md

poodengDev committed:
Fix faq link on contributing.md

poodengDev committed:
Update contributing link aswell to faq

poodengDev committed:
fix broken links

shakkernerd committed:
Merge branch 'develop' into 1838-fix-faq-link-mtoor

juanc07 commented:
What do you mean should be gated? hmmm the twilio auth and sid will be
loaded via vars in .env but sure i'll check if we do need secret because i
think the Auth value is the twilio secret, BTW your twilio account should
be verified, you have a verified twilio phone number, subscription and have
enought credits if you have this you can send sms and whatsapp messages in
any phone number. If you are just using sandbox or free trial then you can
only send sms and whatsapp to your self only

On Sun, Jan 5, 2025, 4:46 AM Patrick ***@***.***> wrote:

> ph represent, this is nice, salamat juan for this!!!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/elizaOS/eliza/pull/1822#issuecomment-2571407525>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALIZMEHEQIACG5PPMBYMTVT2JBCDRAVCNFSM6AAAAABUTKUV6WVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNZRGQYDONJSGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

juanc07 commented:
> ph represent, this is nice, salamat juan for this!!!

thanks but not approved yet, hahaha!

juanc07 committed:
added check gated the plugin-twilio in agent runtime only add it when required env vars are found

tcm390 committed:
sort timelines based on the llm score

juanc07 commented:
suddenly i got integration test issue, but the error is not related to plugin-twilio here is the log "packages/plugin-node postinstall:     throw new ERR_MODULE_NOT_FOUND(
packages/plugin-node postinstall:           ^
packages/plugin-node postinstall: Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/runner/work/eliza/eliza/packages/plugin-node/node_modules/@elizaos/core/dist/index.js' imported from /home/runner/work/eliza/eliza/packages/plugin-node/scripts/postinstall.js"

juanc07 commented:
BTW my PR is behind the develop branch , should i do the rebase or the moderator or admin only?

juanc07 commented on agent/src/index.ts:
added checks now in Agent Runtime it will only include plugin-twilio when vars required for Twilio is found

odilitime replied:
Perfect!

odilitime committed:
Merge branch 'develop' into feature/plugin-twilio

Gajesh2007 committed:
fix: fixed the anthropic regex

Gajesh2007 committed:
Merge branch 'main' of https://github.com/Gajesh2007/eliza

Gajesh2007 committed:
Merge main: Add reclaim adapter and resolve conflicts

Gajesh2007 committed:
fix: conflicts

Gajesh2007 committed:
Merge branch 'develop' into main

aisjdoaisjdoi2123 committed:
Update .env.example

lalalune committed:
Merge pull request #1835 from IterateFast-Labs/develop

Update README_KOR.md

lalalune committed:
Merge pull request #1832 from Valetek/refactor/readme

refactor: typo in readme

lalalune committed:
Merge pull request #1847 from manotoor/1838-fix-faq-link-mtoor

Fix faq link on contributing.md

lalalune committed:
Merge pull request #1787 from RobertSloan22/patch-1

Create README.md

lalalune committed:
Merge pull request #1827 from macfly-base/main

Update c3po.character.json

mcp97 committed:
Merge branch 'develop' into test/goals-memory-provider

aisjdoaisjdoi2123 committed:
Add files via upload

aisjdoaisjdoi2123 committed:
Add files via upload

aisjdoaisjdoi2123 committed:
Merge branch 'develop' into develop

lalalune committed:
Merge branch 'feat/genlayer-plugin' of https://github.com/yeagerai/eliza into HEAD

daniel-farina committed:
Add Stargaze plugin

daniel-farina committed:
Merge branch 'develop' into stargaze-merge-conflicts

lalalune committed:
Merge branch 'bugfix/broken-ci-docs-missed-frozen' of https://github.com/marcellodesales/elizaOS into HEAD

lalalune committed:
Merge branch 'patch-4' of https://github.com/affaan-m/eliza into HEAD

lalalune committed:
Merge branch '1838--fix-faq-link-in-contributing-md' of https://github.com/MacsDickinson/eliza into HEAD

lalalune committed:
Merge branch 'develop' of https://github.com/inferanetwork/eliza into HEAD

lalalune committed:
Merge branch 'main' of https://github.com/robin-rrt/eliza into HEAD

daniel-farina committed:
Merge branch 'develop' into stargaze-merge-conflicts

lalalune committed:
Merge branch 'develop' of https://github.com/blockydevs/eliza into HEAD

lalalune committed:
Merge branch 'roshan/plugin-tee-marlin' of https://github.com/marlinprotocol/eliza into HEAD

lalalune committed:
fix lock

odilitime committed:
Add DAO donation ask & dev discord

odilitime committed:
Merge pull request #1867 from elizaOS/odilitime-patch-1

docs: Add DAO donation ask & dev discord

juanc07 committed:
update repo from develop and fix conflict

juanc07 committed:
added core from main

juanc07 committed:
added prune before installing in integrationTests.yaml file

juanc07 committed:
modify the package.json of elizaos/core added require ./dist/index.js and remove @elizaos/source: ./src/index.ts

juanc07 committed:
use node version 20 for integration test yaml

juanc07 committed:
use node version 23.5.0 for integration test yaml

juanc07 committed:
modify exports  package.json of core and plugin-node

juanc07 committed:
added pnpm build command before post install in plugin node

juanc07 committed:
use plugin-node from main

tcm390 committed:
clean code and add comment

tcm390 committed:
add MAX_ACTIONS_PROCESSING

tcm390 committed:
Merge branch 'main' into tcm-max-actions

tcm390 committed:
remove unused variable

tcm390 committed:
remove unused variable

juanc07 committed:
test remove post install js call

juanc07 committed:
increase version of plugin-node

tcm390 committed:
add EmbeddingModelSettings type

tcm390 committed:
get the info from model settings instead of hardcoced number

juanc07 commented:
i tried all that i can do but it seems some where in the repo the plugin-node is link or pointing to this directory "/home/runner/work/eliza/eliza/packages" so for me it seems like a local machine directory but the weird thing is i can't find it in the project. the whole error is "packages/plugin-node postinstall:     throw new ERR_MODULE_NOT_FOUND(
packages/plugin-node postinstall:           ^
packages/plugin-node postinstall: Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/runner/work/eliza/eliza/packages/plugin-node/node_modules/@elizaos/core/dist/index.js' imported from /home/runner/work/eliza/eliza/packages/plugin-node/scripts/postinstall.js"

tcm390 committed:
remove unused variable

juanc07 committed:
change node version integrationTest.yaml to 23.3.0

tcm390 committed:
fix provider path

tcm390 committed:
Merge branch 'main' into tcm-refactor-modelConfig

tcm390 committed:
Merge branch 'develop' into tcm-refactor-modelConfig

juanc07 committed:
do some cleaning and rebuild

odilitime committed:
Merge branch 'develop' into docs/readme-ro

odilitime committed:
fix bad merge

juanc07 committed:
pmpm-lock changes

juanc07 committed:
added todo in code

odilitime committed:
revert 1834 changes

odilitime committed:
audit fix

shakkernerd committed:
Merge pull request #1872 from elizaOS/fix-postinstall-script

fix: Fix postinstall script

shakkernerd committed:
Merge branch 'develop' into tcm-refactor-modelConfig

odilitime committed:
Merge branch 'develop' into docs/readme-ro

odilitime committed:
Merge pull request #1770 from pedronovikovborges/docs/readme-ro

docs: add Romanian README translation

juanc07 committed:
applied fix from develop branch

tcm390 committed:
resolve conflict

shakkernerd committed:
Merge branch 'develop' into test/goals-memory-provider

juanc07 commented:
the issue about the integration test is now fix, thank you @odilitime

odilitime committed:
Merge branch 'develop' into fix/twitter-profile-cache-1638

shakkernerd committed:
Merge pull request #1840 from ai16z-demirix/test/goals-memory-provider

test: adding more tests for goals,memory and provider. Fixing generation.test.ts

odilitime committed:
Merge branch 'develop' into fix/twitter-profile-cache-1638

odilitime committed:
Merge pull request #1782 from augchan42/fix/twitter-profile-cache-1638

fix: remove twitter profile caching (#1638)

shakkernerd committed:
Merge branch 'develop' into add-devcontainer

shakkernerd committed:
Merge pull request #1807 from jazzvaz/add-devcontainer

feat: Add devcontainer

shakkernerd committed:
Merge branch 'develop' into fix/await-cached-prices-2

shakkernerd committed:
Merge pull request #1828 from bentatum/fix/await-cached-prices-2

fix(solana token provider): await cached data

odilitime committed:
support more debians distros

odilitime committed:
Merge branch 'develop' into fix-postinstall-script

shakkernerd committed:
Merge pull request #1875 from odilitime/fix-postinstall-script

chore: support more debians distros

tcm390 committed:
Merge branch 'develop' into tcm-refactor-modelConfig

shakkernerd committed:
Merge pull request #1805 from elizaOS/tcm-refactor-modelConfig

feat: refactor model config

rawgroundbeef committed:
chore: enable strict null checks

ai16z-demirix committed:
test: adding test setup for telegram client

ai16z-demirix committed:
test: adding test setup for telegram client - improving message manager tests

ai16z-demirix committed:
Merge branch 'develop' into test/client-telegram

YoungPhlo committed:
docs: Add "What Did You Get Done This Week? 8" notes

kylebuildsstuff committed:
Add OpenWeather plugin

Arrowana committed:
Use recommended settings in jupiter swap

JoeyKhd committed:
upd: fixed broken API Documentation URL

JoeyKhd committed:
upd: fixed path

Gajesh2007 committed:
Merge branch 'elizaOS:main' into main

Gajesh2007 committed:
Merge old into develop with regenerated pnpm-lock.yaml

Gajesh2007 committed:
Merge develop into main

Gajesh2007 committed:
fix: settings error

fabianhug committed:
Merge pull request #1558 from Gajesh2007/main

feat: added new plugin - zktls - reclaim

juanc07 committed:
get changes from develop and fix conflict

augchan42 committed:
Merge branch 'elizaOS:main' into fix/twitter-wildcard-targets-1556

HashWarlock committed:
feat: add remote attestation action

augchan42 committed:
just add needed change to run unit test

juanc07 committed:
fixed pnpm-lock file

shakkernerd committed:
chore: pnpm lock file

YoungPhlo committed:
docs: Re-fix timestamp links in WDYGDTW 4 notes

shakkernerd committed:
Merge branch 'develop' into pr-1884

shakkernerd committed:
chore: fix broken pnpm lock file

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
fix: change file extension from rs -> ts

shakkernerd committed:
Merge branch 'develop' into pr-1885

YoungPhlo committed:
docs: Fix timestamp links for Dev School 3

shakkernerd committed:
Merge branch 'develop' into chore/jupiter-swap-recommended-settings

shakkernerd committed:
Merge pull request #1882 from Arrowana/chore/jupiter-swap-recommended-settings

feat: Use recommended settings in jupiter swap

shakkernerd committed:
Merge branch 'develop' into fix-broken-docs-api-url

shakkernerd committed:
Merge pull request #1881 from JoeyKhd/fix-broken-docs-api-url

Fixed broken API Documentation URL

shakkernerd committed:
Merge branch 'develop' into pr-1880

shakkernerd committed:
Merge branch 'develop' into test/client-telegram

shakkernerd committed:
Merge pull request #1879 from ai16z-demirix/test/client-telegram

test: adding test setup for telegram client

YoungPhlo committed:
docs: Fix timestamp links for Dev School 4

shakkernerd committed:
Merge branch 'develop' into pr-1878

YoungPhlo committed:
docs: Re-fix timestamp links in WDYGDTW 5 notes

shakkernerd committed:
fix: conflicting Readme.md files

zhourunlai committed:
fix case sensitive path

shakkernerd committed:
chore: explicitly set node version and bump pnpm version

shakkernerd committed:
Merge branch 'develop' into fix_case_sensitive_path

shakkernerd committed:
Merge pull request #1892 from zhourunlai/fix_case_sensitive_path

fix: remove case sensitive path

shakkernerd committed:
Merge branch 'develop' into fix/tests

shakkernerd committed:
chore: remove pnpm install with no frozen lockfile flag

shakkernerd committed:
chore: pnpm lock file

odilitime committed:
hyperfi endpoint, clean up multer/imports/console logs, notes

shakkernerd committed:
Merge pull request #1896 from odilitime/hfi

feat: hyperfy support

shakkernerd committed:
chore: make viem a global package & remove from plugins package.json

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
chore: add no frozen install flag

juanc07 commented:
after merging with new changes from develop branch i got issues and error in integration test again "Error: ...1816_a2665c1acc2b05267414f68f617d222f npm-install: src/utils/root-ca.ts(4330,14): error TS2742: The inferred type of 'ROOT_CAS' cannot be named without a reference to '.bin/store/v3/tmp/_tmp_1816_a2665c1acc2b05267414f68f617d222f/node_modules/@peculiar/x509'. This is likely not portable. A type annotation is necessary.
...1816_a2665c1acc2b05267414f68f617d222f npm-install: npm error code 1
...1816_a2665c1acc2b05267414f68f617d222f npm-install: npm error path /home/runner/setup-pnpm/node_modules/.bin/store/v3/tmp/_tmp_1816_a2665c1acc2b05267414f68f617d222f
...1816_a2665c1acc2b05267414f68f617d222f npm-install: npm error command failed
...1816_a2665c1acc2b05267414f68f617d222f npm-install: npm error command sh -c npm run build
...1816_a2665c1acc2b05267414f68f617d222f npm-install: npm error A complete log of this run can be found in: /home/runner/.npm/_logs/2025-01-06T01_18_30_870Z-debug-0.log
...1816_a2665c1acc2b05267414f68f617d222f npm-install: Failed"

odilitime committed:
Merge branch 'develop' into docs/community-streams

shakkernerd committed:
chore: remove flag and disable removing lock file

shakkernerd committed:
chore: changes to packages

shakkernerd committed:
chore: add frozen flag

zhourunlai committed:
fix twitter undefined properties

tomguluson92 committed:
update goplus to the safety module

tomguluson92 committed:
Update README.md

RatakondalaArun committed:
fix: add callback to the evaluators for client-telegram

hellopleasures commented:
We have all the same error, maybe the error is on the develop branch.

Commit:
Add Dutch (Nederlands) translation

0xbeekeeper committed:
Update index.ts

lalalune committed:
Merge branch 'develop' into add-dutch-translation

lalalune committed:
Merge pull request #1909 from xzrfax/add-dutch-translation

Add Dutch (Nederlands) translation

lalalune committed:
Merge pull request #1905 from zhourunlai/fix_twitter_undefined_properties

fix: twitter result obj undefined properties

shakkernerd committed:
chore: remove conflict package

shakkernerd committed:
Merge branch 'develop' into fix/tests

shakkernerd committed:
Revert "chore: enable strict null checks"

This reverts commit fcf913797d7345b3e418c817d88347b5ae87868c.

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
Merge pull request #1893 from elizaOS/fix/tests

chore: fix integrations and smoke tests

shakkernerd committed:
Merge branch 'develop' into 1906--fix-evaluate-callback-for-client-telegram

bodhi-crypo committed:
Update summarize_conversation.ts

JoeyKhd committed:
upd: resolved a bunch of type related errors and ensure project compiles properly

JoeyKhd committed:
Merge branch 'develop' into fix-types-and-build

Lukapetro committed:
chore: resolve merge conflicts with develop

rawgroundbeef committed:
fix(core): check evaluators for null

shakkernerd committed:
Merge pull request #1908 from RatakondalaArun/1906--fix-evaluate-callback-for-client-telegram

fix: add callback to the evaluators for client-telegram

shakkernerd committed:
Merge branch 'develop' into fix-types-and-build

shakkernerd committed:
Merge pull request #1916 from bodhi-crypo/main

chore(client-discord): fix SUMMARIZE prompt

shakkernerd committed:
Merge branch 'develop' into fix/runtime-evaluate-null

shakkernerd committed:
Merge pull request #1918 from bentatum/fix/runtime-evaluate-null

fix(core): check evaluators for null

shakkernerd committed:
Merge branch 'develop' into fix-types-and-build

shakkernerd committed:
Merge pull request #1917 from JoeyKhd/fix-types-and-build

fix: resolved a bunch of type related errors and ensure project compiles

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #1907 from tomguluson92/main

feats: diagram update

tcm390 committed:
add followinf option for timeline action

zhourunlai committed:
fix empty input text

tcm390 committed:
Merge branch 'develop' into tcm-max-actions

shakkernerd committed:
Merge pull request #1824 from elizaOS/tcm-max-actions

feat: Optimize Agent Action Processing by Prioritizing Timelines and Limiting Actions Per Cycle

wtfsayo committed:
Merge branch 'develop' into feature/add-binance-plugin

petryshkaCODE committed:
Fix typo Update CHANGELOG.md

wtfsayo committed:
Merge pull request #1812 from Lukapetro/feature/add-binance-plugin

Feature/add binance plugin

wtfsayo committed:
Merge branch 'develop' into Fix-typo

wtfsayo committed:
Merge pull request #1922 from petryshkaCODE/Fix-typo

Fix typo Update CHANGELOG.md

wtfsayo committed:
Merge branch 'develop' into fix_empty_input_text

wtfsayo committed:
Merge branch 'develop' into goplus-security

wtfsayo committed:
Merge pull request #1898 from 0xbeekeeper/goplus-security

feat: add GoPlus Security Plugin to enhance security for agent

JoeyKhd committed:
upd: hide character settings secrets from api route

mcp97 committed:
Merge branch 'develop' into fix_empty_input_text

shakkernerd committed:
Merge branch 'develop' into hide-secrets

shakkernerd committed:
Merge pull request #1923 from JoeyKhd/hide-secrets

API route update

mcp97 committed:
Merge branch 'develop' into fix_empty_input_text

ai16z-demirix committed:
test: adding tests for discord-client

jonathangus committed:
Merge branch 'develop' into dynamic-template

shakkernerd committed:
Merge pull request #1919 from zhourunlai/fix_empty_input_text

fix: handle empty input text to avoid memory content empty error

shakkernerd committed:
Merge branch 'develop' into tests/client-discord

shakkernerd committed:
Merge pull request #1927 from ai16z-demirix/tests/client-discord

test: adding tests for discord-client

shakkernerd committed:
Merge branch 'develop' into docs/community-streams

shakkernerd committed:
Merge pull request #1894 from YoungPhlo/docs/community-streams

docs: Add "What Did You Get Done This Week? #8" notes

proteanx committed:
rm unused variable x_server_url

tcm390 committed:
Limit the number of timelines fetched

tcm390 committed:
make a constant

tcm390 committed:
add comment

sinecose committed:
Merge branch 'develop' into zkevm-no-warning

shakkernerd committed:
Merge pull request #1931 from elizaOS/tcm-limit-fetchingTimelines

fix: Limit the number of timelines fetched

shakkernerd committed:
Merge branch 'develop' into rmxserver

shakkernerd committed:
Merge pull request #1930 from proteanx/rmxserver

fix: rm unused variable X_SERVER_URL

HashWarlock committed:
Merge branch 'develop' into dynamic-template

HashWarlock committed:
Merge pull request #1467 from jonathangus/dynamic-template

feat: Make templates in composeContext dynamic

threewebcode committed:
docs: it is processAction instead of triggerAction

There is only processAction in runtime instance. Update them correctly.

shakkernerd committed:
Merge pull request #1937 from threewebcode/develop

docs: it is processAction instead of triggerAction

Commit:
feat: support for eternalai provider can make request with chain_id extra data in body

odilitime committed:
Merge branch 'develop' of https://github.com/elizaos/eliza into feat/sol-agent-kit

odilitime committed:
Merge pull request #1373 from renlulu/feat/sol-agent-kit

feat: solana token deploy using solana agent kit

Commit:
Merge branch 'develop' into eternal-ai-pr-chainid

bendanzhentan committed:
refactor: remove unused imports

shakkernerd committed:
Merge branch 'develop' into chore/remove-unused-imports

shakkernerd committed:
Merge pull request #1941 from bendanzhentan/chore/remove-unused-imports

chore(plugin-conflux): remove unused imports

shlokkhemani committed:
chore: add embedding tests

ag-wnl committed:
Merge branch 'develop' into zkevm-no-warning

ag-wnl committed:
Merge pull request #1744 from sinecose/zkevm-no-warning

feat(plugin-cronoszkevm): rm not used imports

0xRider committed:
chore: more specific rpc urls

0xRider committed:
Merge branch 'develop' into chore/more-specific-rpc-urls

shakkernerd committed:
Merge pull request #1938 from TrustlessComputer/eternal-ai-pr-chainid

feat: support for eternalai provider can make request with chain_id extra data in body

shakkernerd committed:
Merge branch 'develop' into chore/more-specific-rpc-urls

Commit:
feat: support for eternalai provider can write request/response, change name in env

shakkernerd committed:
Merge pull request #1945 from 0xRider/chore/more-specific-rpc-urls

chore: more specific rpc urls

Commit:
Merge branch 'develop' into eternal-ai-pr-chainid

shakkernerd committed:
Merge pull request #1948 from TrustlessComputer/eternal-ai-pr-chainid

feat: support for eternalai provider can write request/response log info

shakkernerd committed:
Merge branch 'develop' into add-embedding-tests

shakkernerd committed:
Merge pull request #1944 from shlokkhemani/add-embedding-tests

chore: add embedding tests

juanc07 committed:
merged branch with develop branch

juanc07 commented:
integration test is working again after merging with latest develop branch

Freytes commented:
The workflow should be the end user can text texts or call the number and gets the agent. The voice of the agent should use Elevenlabs, if at all possible

odilitime commented:
Hello,

We are changing our plugin development strategy to be more scalable. We have moved the plugins out into their own repos and we're looking for people to either maintain those or own them on their own Github.

If you'd like to be a maintainer,  file an issue in the plugin repo and join our Discord https://discord.gg/elizaos to coordinate.

If you'd like to control the plugin on your own Github, please add an issue to the plugin repo pointing to your repo, and add a modification to the registry. Submit a PR to edit the registry here: https://github.com/elizaos-plugins/registry

Closing this PR for now. Let us know if you have any questions.
